src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Main {
+    private static final FigureSupplier figureSupplier = new FigureSupplier();
+
+    public static void main(String[] args) {
+        AbstractFigure[] figures = new AbstractFigure[6];
+        for (int i = 0; i < figures.length / 2; i++) {
+            figures[i] = (AbstractFigure) figureSupplier.getRandomFigure();
```

Comment:
remove casting to `AbstractFigure`, fix `getRandomFigure()` as returning the `Figure` class